src/function_registry.ts

```diff
@@ -33,12 +33,29 @@ const register = (
   signatureType: SignatureType,
   userFunction: HandlerFunction
 ): void => {
+  if (!isValidFunctionName(functionName)) {
+    throw new Error(`Invalid function name: ${functionName}`);
+  }
+
   registrationContainer.set(functionName, {
     signatureType,
     userFunction,
   });
 };
 
+/**
+ * Returns true if the function name is valid (lowercase alphanumeric or dash characters).
+ * Does not check for beginning or ending with a dash.
+ * Does not check for length <= 63 characters.
+ * @param functionName the function name
+ * @returns true if the function name is valid
+ */
+export const isValidFunctionName = (functionName: string): boolean => {
+  // Validate function name with alpha characters, numbers, and dashes
+  const regex = /^[a-zA-Z-]+$/;
```

Comment:
Can we also impose the restrictions that the function name
- must be <= 63 characters
- must start with a letter 
- must end with a letter or number

Something like:

```
const regex = /[A-Za-z](?:[-_A-Za-z0-9]{0,61}[A-Za-z0-9])?/;
```

Reply:
We can, although I don't know if we should restrict the function name to exactly what a specific vendor requires, so I didn't make the regex too tight.

WDYT?

Reply:
I don't think these limitations are so onerous that they will degrade the usability for non-GCF customers.

Reply:
I disagree that non-Google customer would not want a function name outside of these restrictions, such as 70 characters or non-alphanumeric characters. I believe the FF shouldn't restrict its usage for the limitations of one specific platform.

However, I've implemented your suggestion as it doesn't matter too much.

Reply:
Thank you!